File: arrodashu-server/src/main/java/com/arrodashu/entity/Species.java
package com.arrodashu.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

/**
 * 品种实体
 */
@Data
@TableName("tree_species")
public class Species {
    
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private String name;
    
    private String latinName;
    
    private String alias;
    
    private Long familyId;
    
    private Long genusId;
    
    // 形态特征
    private Integer treeType;
    
    private Integer heightMin;
    
    private Integer heightMax;
    
    private String crownShape;
    
    private String barkDesc;
    
    private String leafDesc;
    
    private String flowerDesc;
    
    private String fruitDesc;
    
    // 生长习性
    private String lightPreference;
    
    private String waterPreference;
    
    private String soilPreference;
    
    private String hardinessZone;
    
    private String growthRate;
    
    // 园林用途
    private String landscapeUse;
    
    // 其他
    private String description;
    
    private String origin;
    
    private Long createBy;
    
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;
    
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
    
    private Integer status;
    
    // 非数据库字段
    @TableField(exist = false)
    private String familyName;
    
    @TableField(exist = false)
    private String genusName;
    
    @TableField(exist = false)
    private List<String> images;
    
    @TableField(exist = false)
    private List<Long> tagIds;
}

File: arrodashu-web/src/api/species.ts
import request from './request'

export interface Species {
  id: number
  name: string
  latinName: string
  alias?: string
  familyId: number
  genusId: number
  familyName?: string
  genusName?: string
  treeType?: number
  heightMin?: number
  heightMax?: number
  lightPreference?: string
  waterPreference?: string
  soilPreference?: string
  hardinessZone?: string
  growthRate?: string
  landscapeUse?: string
  description?: string
  origin?: string
  status: number
  createTime?: string
}

export interface PageResult<T> {
  records: T[]
  total: number
  size: number
  current: number
  pages: number
}

export interface SpeciesQuery {
  page?: number
  size?: number
  keyword?: string
  familyId?: number
  genusId?: number
}

// 获取品种列表
export function getSpeciesList(params: SpeciesQuery) {
  return request.get<any, { data: PageResult<Species> }>('/species', { params })
}

// 获取品种详情
export function getSpeciesDetail(id: number) {
  return request.get<any, { data: Species }>(`/species/${id}`)
}

// 添加品种
export function addSpecies(data: Species) {
  return request.post('/species', data)
}

// 更新品种
export function updateSpecies(id: number, data: Species) {
  return request.put(`/species/${id}`, data)
}

// 删除品种
export function deleteSpecies(id: number) {
  return request.delete(`/species/${id}`)
}
